libraries/AC_Avoidance/AP_Astar_RT.cpp

```diff
@@ -0,0 +1,686 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
```

Comment:
It might be helpful to put a high level description of how the library works here or in the .h file.  So perhaps list out the stages that it goes through to calculate the path and name the methods involved specifically.